packages/appsync-modelgen-plugin/src/utils/process-connections-v2.ts

```diff
@@ -0,0 +1,103 @@
+import { CodeGenField, CodeGenFieldDirective, CodeGenModel, CodeGenModelMap } from '../visitors/appsync-visitor';
+import {
+  CodeGenFieldConnection, DEFAULT_HASH_KEY_FIELD, flattenFieldDirectives,
+  getDirective,
+  makeConnectionAttributeName,
+} from './process-connections';
+import { processHasOneConnection } from './process-has-one';
+import { processBelongsToConnection } from './process-belongs-to';
+import { processHasManyConnection } from './process-has-many';
+
+// TODO: This file holds several references to utility functions in the v1 process connections file, those functions need to go here before that file is removed
+
+export function getConnectedFieldV2(field: CodeGenField, model: CodeGenModel, connectedModel: CodeGenModel, directiveName: string): CodeGenField {
```

Comment:
Originally I was splitting it up in that way but I was finding that for this function, the majority of the code was being duplicated from function to function, so I felt like it made more sense to leave it as one. I can still go back and change that if we'd rather take that approach, but it will mean a lot of code duplication